Add support for Memcached ElastiCache cache cluster

hectcastro opened the pull request:
Add the core components necessary for a Memcached ElastiCache cache cluster:

- Variable sized ElastiCache cache cluster
- Cache cluster security group with no rules
- CloudWatch Alarms for CPU and memory utilization

notthatbreezy commented on outputs.tf:
```diff
@@ -0,0 +1,3 @@
+output "configuration_endpoint" {
+  value = "${aws_elasticache_cluster.memcached.configuration_endpoint}"
```
Think it would be good to output the security group ID here as well so it can be modified by external roles if necessary

hectcastro replied:
Addressed in https://github.com/azavea/terraform-aws-memcached-elasticache/commit/e6555bb497469760ecaa4e70d295a903548899f5.

notthatbreezy commented on main.tf:
```diff
@@ -0,0 +1,89 @@
+resource "aws_security_group" "memcached" {
+  vpc_id = "${var.vpc_id}"
+
+  ingress {
```
I think the ingress/egress rules should be split out (as per the note [here](https://www.terraform.io/docs/providers/aws/r/security_group.html)) if we want to put the security group ID in outputs; otherwise dependent terraform roles could see unexpected behavior due to rule overwrites.

notthatbreezy commented:
Cool, +1

Commit:
Add support for Memcached ElastiCache cache cluster

Add the core components necessary for a Memcached ElastiCache cache cluster:

- Variable sized ElastiCache cache cluster
- Cache cluster security group with no rules
- CloudWatch Alarms for CPU and memory utilization

notthatbreezy commented on variables.tf:
```diff
@@ -0,0 +1,46 @@
+  default = "75"
+}
+
+variable "alarm_memory_threshold" {
```
Minor nitpick, on these types of variables I've found it easier to parse the values if the name reflects the unit. In this case maybe `alarm_memory_threshold_bytes`. It avoids having to read the comment or add comments for the variable to make sense in tfvar files then.

hectcastro replied:
This is a good point. Here I was trying to mimic the variable names for the Redis module, but I'll update that here and create an issue for the Redis module to copy it.

notthatbreezy reviewed (approved):
Looks good, one minor comment re: variable names
